odoo/addons/base/models/qweb.py

```diff
@@ -143,16 +143,23 @@ def _compile(self, template, options):
         if not ref:
             ref = element.get('t-name', str(document))
 
+        # reference to get xml and etree (usually the template ID)
         options['ref'] = ref
-        if 'profile' in options:
-            options['document'] = document if isinstance(document, str) else str(document, 'utf-8')
+        # str xml of the reference template used for compilation. Useful for debugging, dev mode and profiling.
+        options['ref_xml'] = document if isinstance(document, str) else str(document, 'utf-8')
 
         _options = dict(options)
         options = frozendict(options)
 
+        # Initial template value send to render method (not in the froozen dict because it may be
+        # different from one render to another. Indeed, it may be the view ID or the key)
         _options['template'] = template
+        # Root of the etree which will be processed during compilation.
         _options['root'] = element.getroottree()
+        # Reference to the last node being compiled. It is mainly used for debugging and displaying
```

Comment:
Can you add this comments in the first commit with other non profiling related changes?